File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.py
# 아홉 명의 난쟁이 키를 입력받습니다.
dwarfs = [int(input().strip()) for _ in range(9)]

# 아홉 명의 난쟁이 키의 합을 구합니다.
total_sum = sum(dwarfs)

# 키의 합에서 100을 뺀 값을 구합니다.
target = total_sum - 100

# 두 명의 난쟁이 키를 선택하여 그 합이 target과 같은지 확인합니다.
found = False
for i in range(8):
    for j in range(i + 1, 9):
        if dwarfs[i] + dwarfs[j] == target:
            dwarf1, dwarf2 = dwarfs[i], dwarfs[j]
            found = True
            break
    if found:
        break

# 선택된 두 난쟁이를 제외한 나머지 일곱 난쟁이 키를 구합니다.
remaining_dwarfs = [d for d in dwarfs if d != dwarf1 and d != dwarf2]

# 일곱 난쟁이 키를 오름차순으로 정렬하여 출력합니다.
remaining_dwarfs.sort()
for dwarf in remaining_dwarfs:
    print(dwarf)

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int[] dwarfs = new int[9];
        for (int i = 0; i < 9; i++) {
            dwarfs[i] = Integer.parseInt(br.readLine());
        }
        
        int totalSum = 0;
        for (int dwarf : dwarfs) {
            totalSum += dwarf;
        }
        
        int target = totalSum - 100;
        int dwarf1 = 0, dwarf2 = 0;
        boolean found = false;
        
        for (int i = 0; i < 8; i++) {
            for (int j = i + 1; j < 9; j++) {
                if (dwarfs[i] + dwarfs[j] == target) {
                    dwarf1 = dwarfs[i];
                    dwarf2 = dwarfs[j];
                    found = true;
                    break;
                }
            }
            if (found) break;
        }
        
        int[] remainingDwarfs = new int[7];
        int index = 0;
        for (int dwarf : dwarfs) {
            if (dwarf != dwarf1 && dwarf != dwarf2) {
                remainingDwarfs[index++] = dwarf;
            }
        }
        
        Arrays.sort(remainingDwarfs);
        for (int dwarf : remainingDwarfs) {
            System.out.println(dwarf);
        }
    }
}
